File: mobile-wallet-master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/base/BaseActivityCallback.java
package org.mifos.mobilewallet.mifospay.base;

import android.support.v4.widget.SwipeRefreshLayout;

public interface BaseActivityCallback {

    void showSwipeProgress();

    void hideSwipeProgress();

    void showProgressDialog(String message);

    void hideProgressDialog();

    void setToolbarTitle(String title);

    void setSwipeRefreshEnabled(boolean enabled);

    void showBackButton();

    void hideBackButton();

    SwipeRefreshLayout getSwipeRefreshLayout();
}

File: mifospay/src/main/java/org/mifospay/base/BaseActivityCallback.kt
/*
 * Copyright 2024 Mifos Initiative
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * See https://github.com/openMF/mobile-wallet/blob/master/LICENSE.md
 */
package org.mifospay.base

import androidx.swiperefreshlayout.widget.SwipeRefreshLayout

interface BaseActivityCallback {
    fun showSwipeProgress()
    fun hideSwipeProgress()
    fun showProgressDialog(message: String?)
    fun hideProgressDialog()
    fun cancelProgressDialog()
    fun setToolbarTitle(title: String?)
    fun setSwipeRefreshEnabled(enabled: Boolean)
    fun showColoredBackButton(drawable: Int)
    fun showCloseButton()
    fun hideBackButton()
    fun dismissProgressDialog()

    val swipeRefreshLayout: SwipeRefreshLayout?
}
